database/db_map.py

```diff
@@ -43,4 +41,74 @@ def get_receive_stations(self, rcv_station_dict):
                                                     categories: list(int),
                                                     items: list(int)}, ...}
         """
-        raise NotImplementedError
+        response = self.db.get("receiver_stations", None)
+        self.checking(rcv_station_dict, response)
+        return response
+
+    def checking(self, user_station, stations):
+        """
+        Return all suitable for a user stations,
+        otherwise return []
+        :param user_station: dict = {(str): int, (str): (str), ...}
+        :param stations:  dict = {(str): list, (str): (int), (str): (str), ...}
+        :return: list = [(dict), ...]
+        """
+        suitable_stations = []
+        for station in stations:
+            if self._data_intersection(user_station['categories'], station['categories']):
+                if self._data_intersection(user_station['city_id'], station['locations']):
```

Comment:
You are comparing here city_id and location_id. Which is not really what is needed.
Create function `get_location(location_id)` from which you can extract location info with city_id.